sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -442,7 +463,7 @@ def __init__(
     ):
         self._read_buffer = BytesIO()
         self.socket_lock = asyncio.Lock()
-        self.sslopts = self._build_ssl_opts(ssl_opts) if isinstance(ssl_opts, dict) else None
+        self.sslopts = ssl_opts if isinstance(ssl_opts, dict) else None
```

Comment:
building here causes the error to surface from client.open() rather than client.connect(), which causes error handling discrepancy b/w pyamqp and uamqp. updated for parity.

Reply:
comment added to code.